packages/driver/cypress/e2e/commands/assertions.cy.js

```diff
@@ -416,34 +416,40 @@ describe('src/cy/commands/assertions', () => {
       it('does not log extra commands on fail and properly fails command + assertions', function (done) {
         cy.on('fail', (err) => {
           assertLogLength(this.logs, 6)
+          expect(err.message).to.eq('You must provide a valid number to a `length` assertion. You passed: `asdf`')
 
           expect(this.logs[3].get('name')).to.eq('get')
-          expect(this.logs[3].get('state')).to.eq('failed')
-          expect(this.logs[3].get('error')).to.eq(err)
+          expect(this.logs[3].get('state')).to.eq('passed')
+          expect(this.logs[3].get('error')).to.be.undefined
 
           expect(this.logs[4].get('name')).to.eq('assert')
           expect(this.logs[4].get('state')).to.eq('failed')
-          expect(this.logs[4].get('error').name).to.eq('AssertionError')
+          expect(this.logs[4].get('error').name).to.eq('CypressError')
+          expect(this.logs[4].get('error')).to.eq(err)
```

Comment:
| Before | After |
| -- | -- |
|<img width="458" alt="Screen Shot 2022-11-02 at 4 53 54 PM" src="https://user-images.githubusercontent.com/14099737/199610795-4ccc870e-159d-4dcd-9220-67c7117876a5.png">|<img width="557" alt="Screen Shot 2022-11-02 at 5 00 15 PM" src="https://user-images.githubusercontent.com/14099737/199610799-0a639256-7111-423a-8bda-cac776b25338.png">|

Reply:
If the assertion fails, I would expect an `AssertionError` to be thrown here and not a `CypressError`. What was the rason that changed? Or is it that the argument into the assertion is bad?

Reply:
I thought that at first too, but this is one of our own errors:
- thrown here https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/chai.ts#L375
- error message: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/error_messages.ts#L178